### PR TITLE
Dynamic profile filter fields

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

When changing a profile filter type the match field will be changed to suit the filters type's needs.

The following filter types are now dynamic types
- Switch
- Roles
- Connection Types
- Connection Sub Types
## Impacts

The Admin GUI Profiles page
## Dependencies

List the PRs or other factors on which this PR depends
## NEWS file entries

New Features
++++++++++++
- Profile filters provides the appropriate values according its type
## Delete branch after merge

NO

### CODE REVIEW

**Comment by julsemaan** [comment]
me like this

**Comment by jrouzierinverse** [comment]
@lzammit Brought up a point about the Realm profile filter not having a drop down list.
I just want to make sure if it make sense to have a drop down list for realms also.

**Comment by julsemaan** [comment]
I'm not sure...

Is there a way to switch from dropdown to manual ?

**Comment by jrouzierinverse** [comment]
We could create a typeahead for the realms allowing someone to see the configured realms but also allow them to put in their own user defined realm.

**Comment by jrouzierinverse** [comment]
Spoke to King Louis he said we should not worry about this. 

image

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Profile.pm [hunk 1]
```diff
@@ -356,7 +373,7 @@ Returns all the merge paths
 
 sub mergedPaths {
     my ($self, $c) = @_;
-    return (catfile($captiveportal_profile_templates_path, $c->stash->{id}), $self->parentPaths($c));
+    return grep { -d } (catfile($captiveportal_profile_templates_path, $c->stash->{id}), $self->parentPaths($c));
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Profile.pm:376]: Does this belong here ?
Just making sure

Comment (rouzier) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Profile.pm:376]: I found a bug while working on the branch so I fixed it.

**Pull request merged**